pgmpy/models/MarkovModel.py

```diff
@@ -676,3 +676,30 @@ def get_partition_function(self):
             raise ValueError('Factor for all the random variables not defined.')
 
         return np.sum(factor.values)
+
+    def copy(self):
+        """
+        Returns a deep copy of this Markov Model.
+ 
+        Returns
+        -------
+        MarkovModel: Deep copy of this Markov model.
```

Comment:
This is misleading. This is not a deepcopy , nor we have overwritten the copy.deepcopy .
https://docs.python.org/2/library/copy.html